Docker/update_score.py

```diff
@@ -15,9 +24,40 @@ def get_args():
                         help="Resulting scores")
     parser.add_argument("-f", "--all_scores", required=True,
                         help="A csv table collected all submssion scores")
+    parser.add_argument("-s", "--submission_view_synapseid",
+                        required=True, help="Synapse ID of submission view")
+    parser.add_argument("-l", "--leader_board_synapseid",
+                        required=True, help="Synapse ID of leader board")
     return parser.parse_args()
 
 
+def valid_scores(df, col):
+    """Filter out the invalid scores."""
+    # not necessary for live queue
+    valid_inx = []
+    valid_scores = []
+
+    for loc, res in enumerate(df[col]):
+        if len(res) > 0:
+            scores = res[0][1:-1].split(", ")
+            if len(scores) == 8:
+                valid_inx.append(loc)
+                valid_scores.append([float(x) for x in scores])
+    return {"loc": valid_inx, "scores": valid_scores}
+
+
+def rank(l):
+    """Rank the item at the same index across multiple lists."""
+    # get rank a list of lists for each index
+    rank_list = []
+    for i in range(len(l[0])):
+        ranks = rankdata([x[i] for x in l])
+        rank_list.append(list(ranks))
+    # average of ranks for each list
+    avg_rank = [(sum(r)/len(r)) for r in zip(*rank_list)]
+    return avg_rank
```

Comment:
@vpchung
To explain a bit what the function is doing, let's take an example. For instances, we have two lists of scores: [[1,2,3], [2,3,1]] for three test sets. To get a average rank of submission, we will do:
1. rank the score at same index of each list (rank 1 -> smallest score), like: rank(1,2), rank(2,3), rank(3,1)
2. we will have a ranked list for each index position: [[1,2], [1,2], [2,1]]
3. to get average ranks for **each submission**, do average on same index of each index of ranked list ^, like: mean(1,1,2) and mean(2,2,1)
4. we will get a list of averaged ranks for all submissions(input list `l`)

Reply:
Sounds reasonable to me.